### PR TITLE
images work

### PR DESCRIPTION
**Pull request by andredubovski**
Line breaks in a single slack message are now separated by newlines in the markdown

Whenever a message with an image (or any other attachment) is clipped, clipper makes it publicly permalinked through Slack API and includes a link to the attachment(s) in the markdown

Code snippets in Slack are converted to <code> in markdown

### CODE REVIEW

File: slack-clipper-backend/app.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def get_conversation_thread(client, channel, ts):
     users = {}
     for message in replies["messages"]:
         print(message)
+        print("\n")
```
Comment (yirenlu92) [inline slack-clipper-backend/app.py]: please remove print statements from code before checking in

File: slack-clipper-backend/app.py [hunk 2]
```diff
@@ -112,14 +124,15 @@ def upload_markdown_file_to_firebase(markdown_text, userid, channelid):
...
-    print("tried to send a message")
 
     os.remove(filename)
     print(f"File uploaded successfully: {filename}")
 
 
 @app.shortcut("clip_markdown")
 def clip_markdown(ack, shortcut, client):
+
+    print("clipping markdown")
```
Comment (yirenlu92) [inline slack-clipper-backend/app.py:133]: remove print statements

File: slack-clipper-backend/app.py [hunk 3]
```diff
@@ -36,38 +37,50 @@ def get_conversation_thread(client, channel, ts):
...
 
 
-def make_markdown_message(name, profile_picture, text, files):
-    returnString = f"""
+def make_markdown_message(name, profile_picture, text):
+
+    while "```" in text:
+        text=text.replace("```", "<code>", 1)
+        text = text.replace("```", "</code>", 1)
+    text=text.replace("\n", "<div>")
```
Comment (yirenlu92) [inline slack-clipper-backend/app.py]: you don't need to replace some new lines with the closing div tag?

File: slack-clipper-backend/app.py [hunk 4]
```diff
@@ -36,38 +37,50 @@ def get_conversation_thread(client, channel, ts):
...
-            "files": message["files"]
+            "profile_picture": users[user_id]["profile_picture"]
         })
 
+        if "files" in message:
+            permalinksToAttachments = ""
+            for file in message["files"]:
+                if not file['public_url_shared']:
+                    app.client.files_sharedPublicURL(file=file["id"], token=os.environ.get("USER_TOKEN"))
+                permalinksToAttachments += '<a href="'+file["permalink_public"]+'">Download</a><div>'
```
Comment (yirenlu92) [inline slack-clipper-backend/app.py]: can we use String formatting and not concatenation

https://realpython.com/python-string-formatting/#2-new-style-string-formatting-strformat

Comment (yirenlu92) [reply slack-clipper-backend/app.py]: That will make this section a bit easier to read

File: slack-clipper-backend/app.py (hunk 3 above)
Comment (yirenlu92) [reply slack-clipper-backend/app.py]: why are we replacing the markdown code backticks? I think they should render properly in the .md file, no?

File: slack-clipper-backend/app.py (hunk 2 above)
Comment (andredubovski) [reply slack-clipper-backend/app.py:133]: ✅

File: slack-clipper-backend/app.py (hunk 3 above)
Comment (andredubovski) [reply slack-clipper-backend/app.py]: just tested it, the backticks don't render the code properly for some reason in both markdown viewers I tried.

File: slack-clipper-backend/app.py (hunk 4 above)
Comment (andredubovski) [reply slack-clipper-backend/app.py]: ✅

File: slack-clipper-backend/app.py (hunk 3 above)
Comment (yirenlu92) [reply slack-clipper-backend/app.py]: can you send me the markdown file?

**Review by yirenlu92 — APPROVED** [review]

**Pull request merged**